src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/service/SubmissionServiceImpl.java

```diff
@@ -118,10 +122,17 @@ public SubmitResponse createSubmission(Submission submission, AccountDetails acc
                 .submitFilingPackage(
                         accountDetails,
                         submission.getFilingPackage(),
-                        efilingPayment -> paymentAdapter.makePayment(efilingPayment));
-
-        result.setPackageRef(Base64.getEncoder().encodeToString(submitPackageResponse.getPackageLink().getBytes()));
-
+                        paymentAdapter::makePayment);
+
+        if(isEarlyAdopter) {
+            result.setPackageRef(
+                    Base64.getEncoder().encodeToString(
+                            MessageFormat.format("{0}/packagereview/{1}",
+                                    navigationProperties.getBaseUrl(),
+                                    submitPackageResponse.getTransactionId()).getBytes()));
```

Comment:
Are these ever possibly null?  May have an NPE otherwise.

Reply:
They shouldn't be at this point.